zcash_client_sqlite/src/lib.rs

```diff
@@ -795,6 +798,18 @@ impl<P: consensus::Parameters> WalletCommitmentTrees for WalletDb<rusqlite::Conn
             .map_err(|e| ShardTreeError::Storage(commitment_tree::Error::Query(e)))?;
         Ok(())
     }
+
+    fn get_checkpoint_depth(
+        &self,
+        min_confirmations: NonZeroU32,
+    ) -> Result<usize, ShardTreeError<Self::Error>> {
+        get_checkpoint_depth(&self.conn, min_confirmations)
+            .map_err(|e| ShardTreeError::Storage(commitment_tree::Error::Query(e)))?
+            // `CheckpointPruned` is perhaps a little misleading; in this case it's that
+            // the chain tip is unknown, but if that were the case we should never have been
+            // calling this anyway.
```

Comment:
This seems like a poor argument. Just add another error case, if that's not too difficult.

Reply:
It would require a new release of `shardtree`

Reply:
Ok that's a good reason not to.